File: library/src/main/java/me/shkschneider/skeleton/ui/LoadingProgressBar.java
package me.shkschneider.skeleton.ui;

import android.content.Context;
import android.support.v4.widget.ContentLoadingProgressBar;
import android.util.AttributeSet;

// android:progressTint=""
// style="?android:attr/progressBarStyleLarge"
public class LoadingProgressBar extends ContentLoadingProgressBar {

    public LoadingProgressBar(final Context context) {
        this(context, null);
    }

    public LoadingProgressBar(final Context context, final AttributeSet attrs) {
        super(context, attrs);

        setIndeterminate(true);
    }

}

File: library/src/main/java/me/shkschneider/skeleton/ui/LoadingProgressBar.kt
package me.shkschneider.skeleton.ui

import android.content.Context
import android.content.res.ColorStateList
import androidx.core.graphics.drawable.DrawableCompat
import androidx.core.widget.ContentLoadingProgressBar
import android.util.AttributeSet
import androidx.annotation.ColorInt

// android:progressTint=""
// style="?android:attr/progressBarStyleLarge"
class LoadingProgressBar : ContentLoadingProgressBar {

    init {
        isIndeterminate = true
    }

    constructor(context: Context, attrs: AttributeSet? = null) : super(context, attrs)

    fun color(@ColorInt color: Int) {
        if (isIndeterminate) {
            DrawableCompat.setTintList(indeterminateDrawable, ColorStateList.valueOf(color))
        } else {
            DrawableCompat.setTintList(progressDrawable, ColorStateList.valueOf(color))
        }
    }

}
